File: AsphaltRoad.java
import java.util.ArrayList;
import java.util.List;


class AsphaltRoad {
    private List<Vehicle> vehicles;

    public AsphaltRoad() {
        vehicles = new ArrayList<>();
    }

    public void addVehicle(Vehicle vehicle) {
        vehicles.add(vehicle);
    }

    public void fillUp(Vehicle vehicle, int liters) {
        vehicle.fillUp(liters);
    }

    public void drive(Vehicle vehicle, double distance) {
        vehicle.drive(distance);
    }

    public void printVehicleRemainingFuels() {
        for (Vehicle vehicle : vehicles) {
            System.out.println("Vehicle remaining fuel: " + vehicle.remainingFuel());
        }
    }
}

File: asphalt_road.py
from typing import List
from vehicle import Vehicle

class AsphaltRoad:
    def __init__(self):
        self.vehicles: List[Vehicle] = []

    def add_vehicle(self, vehicle: Vehicle):
        self.vehicles.append(vehicle)

    def fill_up(self, vehicle: Vehicle, liters: int):
        vehicle.fill_up(liters)

    def drive(self, vehicle: Vehicle, distance: float):
        vehicle.drive(distance)

    def print_vehicle_remaining_fuels(self):
        for vehicle in self.vehicles:
            print(f"Vehicle remaining fuel: {vehicle.remaining_fuel()}") 